File: src/main/webapp/src/app/models/action-trigger-midi-control-change.ts
import { ActionTriggerMidi } from "./action-trigger-midi";

export class ActionTriggerMidiControlChange extends ActionTriggerMidi {
  controller: number;
  value: number;

  constructor(data?: any) {
    super(data);

    if (!data) {
      return;
    }

    this.controller = data.controller;
    this.value = data.value;

    if (!this.controller) {
      this.controller = 0;
    }
  }

  toJSON() {
    return { actionTriggerMidiControlChange: { ...this } };
  }
}

File: src/main/java/com/ascargon/rocketshow/midi/ActionTriggerMidiControlChange.java
package com.ascargon.rocketshow.midi;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ActionTriggerMidiControlChange extends ActionTriggerMidi {

    // The controller number. If null -> all controllers
    private Integer controller;

    // The controller value. If null -> any value
    private Integer value;

}
